### PR TITLE
Bug fixes and memory/exection time optimizations for large graphs.

### PR DESCRIPTION
**Pull request by feldergast**
- Fixed a big in objectComms where the serialization size was stored in a 32-bit int.
- Changed ConfigGraph so it stored pointers to ConfigComponents instead of the actual objects.
  - Made it so that SparseVectorMap could store pointers
- Added global parameters.  You can now specify a named global set of parameters.  When adding to Param objects, only a pointer to the set will be kept.

### CODE REVIEW

File: src/sst/core/params.h [hunk 1]
```diff
@@ -191,6 +191,8 @@ class Params : public SST::Core::Serialization::serializable {
      * Enable or disable parameter verification on an instance
      * of Params.  Useful when generating a new set of Params to
      * pass off to a module.
+     *
+     * @return returns the previous state of the flay
```
Comment (gvoskuilen) [inline src/sst/core/params.h:195]: flay -> flag

**Comment by nmhamster** [comment]
@feldergast - does this fix the serialization issues we have seen recently?

**Comment by feldergast** [comment]
Yes, it does fix the serialization issue.

**Comment by nmhamster** [comment]
> Yes, it does fix the serialization issue.

Great!

**Pull request merged**